v1/remote/transport/bearer.go

```diff
@@ -41,7 +43,11 @@ func (bt *bearerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 	if err != nil {
 		return nil, err
 	}
-	in.Header.Set("Authorization", hdr)
+
+	// Avoid setting Authorization header on the wrong domain (e.g. on redirect).
```

Comment:
I would lead with something like: `http.Client` handles redirects at a layer above the `http.RoundTripper` abstraction, so to avoid forwarding Authorization headers to places we are redirected, only set it when the authorization header matches the registry with which we are interacting.